src/main/java/org/broadinstitute/hellbender/utils/bigquery/BigQueryUtils.java

```diff
@@ -0,0 +1,325 @@
+        final List<Integer> columnWidths = new ArrayList<>(result.getSchema().getFields().size());
+
+        // Start with schema names:
+        for ( final Field field : result.getSchema().getFields() ) {
+            columnWidths.add( field.getName().length() );
+        }
+
+        // Check each row and each row's array values (if applicable):
+        for ( final FieldValueList row : result.iterateAll() ) {
+            for ( int i = 0; i < row.size() ; ++i ) {
+                // Only get the row size if it's not null:
+                if ( !row.get(i).isNull() ) {
+                    if ( row.get(i).getAttribute() == FieldValue.Attribute.PRIMITIVE ) {
+                        if ( columnWidths.get(i) < row.get(i).getStringValue().length() ) {
+                            columnWidths.set(i, row.get(i).getStringValue().length());
+                        }
+                    }
+                    else {
+                        for ( int j = 0; j < row.get(i).getRepeatedValue().size(); ++j ) {
+                            final String stringValue;
+                            if ( row.get(i).getRepeatedValue().get(j).getAttribute() == FieldValue.Attribute.PRIMITIVE ) {
+                                stringValue = row.get(i).getRepeatedValue().get(j).getStringValue();
+                            }
+                            else {
+                                stringValue = row.get(i).getRepeatedValue().get(j).getRecordValue().get(0).getStringValue();
+                            }
+                            if ( columnWidths.get(i) < stringValue.length() ) {
+                                columnWidths.set(i, stringValue.length());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return columnWidths;
+    }
+
+    /**
+     * Executes the given {@code queryJobConfiguration} on the given {@code bigQuery} instance.
+     * @param bigQuery The instance of {@link BigQuery} to use to connect to BigQuery.
+     * @param queryJobConfiguration The {@link QueryJobConfiguration} object containing all required information to retrieve data from a BigQuery table.
+     * @return A {@link TableResult} object containing the results of the query executed.
+     */
+    private static TableResult submitQueryAndWaitForResults( final BigQuery bigQuery,
+                                                             final QueryJobConfiguration queryJobConfiguration ) {
+        // Create a job ID so that we can safely retry:
+        final JobId jobId = JobId.of(UUID.randomUUID().toString());
+
+        logger.info("Sending query to server...");
+        Job   queryJob       = bigQuery.create(JobInfo.newBuilder(queryJobConfiguration).setJobId(jobId).build());
+
+        // Wait for the query to complete.
+        try {
+            logger.info("Waiting for query to complete...");
+            queryJob = queryJob.waitFor();
```

Comment:
you might want to configure timeouts for this, although it has defaults they may not be what we want